sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionals.scala

```diff
@@ -312,3 +312,63 @@ case class CaseKeyWhen(key: Expression, branches: Seq[Expression]) extends CaseW
     }.mkString
   }
 }
+
+case class Least(children: Expression*)
+  extends Expression {
+
+  override def nullable: Boolean = children.forall(_.nullable)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (children.map(_.dataType).distinct.size > 1) {
+      TypeCheckResult.TypeCheckFailure(
+        s"differing types in Least (${children.map(_.dataType)}).")
+    } else {
+      TypeCheckResult.TypeCheckSuccess
+    }
+  }
+
+  override def dataType: DataType = children.head.dataType
+
+  override def eval(input: InternalRow): Any = {
+    val cmp = GreaterThan
```

Comment:
Oh, sorry, I just notice it's impossible, but after checking the code in `GreateThan`, we can borrow the `ordering = TypeUtils.getOrdering(left.dataType)`